### PR TITLE
feat(nuxt): allow dynamic `scrollToTop` page meta

### LINKED ISSUE #21738: Allow scrollToTop page meta be dynamic with a function that receives the destination route
### Describe the feature

Currently, `scrollToTop` can only be a boolean. Making it a function allows users to turn the auto-scrolling on/off dynamically based on different route parameters.

For example, when you have a catch-all route and you need to handle scrolling yourself on a specific route subpath, this feature will allow us to do that. 

```TS
definePageMeta({
  scrollToTop: (route) => !route.path.startsWith('/handle-scroll-myself')
})
```

### Additional information

- [X] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

### PR DESCRIPTION
**Pull request by prashantpalikhe**
### 🔗 Linked issue

#21738 

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This PR allows `scrollToTop` page meta to be a function besides a boolean. The function receives the destination route object as an argument. 

This will allow users to dynamically set `scrollToTop` based on the route's params, query string, or path.

Resolves #21738 

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/3.api/3.utils/define-page-meta.md [hunk 1]
```diff
@@ -32,6 +32,7 @@ interface PageMeta {
   keepalive?: boolean | KeepAliveProps
   layout?: false | LayoutKey | Ref<LayoutKey> | ComputedRef<LayoutKey>
   middleware?: MiddlewareKey | NavigationGuard | Array<MiddlewareKey | NavigationGuard>
+  scrollToTop?:boolean | ((route: RouteLocationNormalizedLoaded) => boolean)
```
Comment (danielroe) [inline docs/3.api/3.utils/define-page-meta.md:35]:
```suggestion
  scrollToTop?: boolean | ((route: RouteLocationNormalizedLoaded) => boolean)
```

**Review by danielroe — APPROVED** [review]

**Comment by danielroe** [comment]
Beautiful 👌

**Comment by IsraelOrtuno** [comment]
Really nice feature. I was wondering if passing both `from` and `to` routes may be useful so you could make the decision to scroll to top with even more information, thoughts?

**Comment by prashantpalikhe** [comment]
> Really nice feature. I was wondering if passing both `from` and `to` routes may be useful so you could make the decision to scroll to top with even more information, thoughts?

That would definitely be quite useful. Wdyt @danielroe

**Comment by danielroe** [comment]
I agree 👍

**Comment by prashantpalikhe** [comment]
Updated the PR @danielroe

File: docs/3.api/3.utils/define-page-meta.md [hunk 2]
```diff
@@ -32,6 +32,7 @@ interface PageMeta {
   keepalive?: boolean | KeepAliveProps
   layout?: false | LayoutKey | Ref<LayoutKey> | ComputedRef<LayoutKey>
   middleware?: MiddlewareKey | NavigationGuard | Array<MiddlewareKey | NavigationGuard>
+  scrollToTop?:boolean | ((to: RouteLocationNormalizedLoaded, from: RouteLocationNormalizedLoaded) => boolean)
```
Comment (danielroe) [inline docs/3.api/3.utils/define-page-meta.md:35]:
```suggestion
  scrollToTop?: boolean | ((to: RouteLocationNormalizedLoaded, from: RouteLocationNormalizedLoaded) => boolean)
```

**Comment by danielroe** [comment]
Great! Thank you; we'll slot this in for the 3.7 release ❤️

**Review by danielroe — APPROVED** [review]

**Pull request merged**